interfaces/builtin/utils.go

```diff
@@ -110,7 +111,20 @@ func aareExclusivePatterns(orig string) []string {
 // dirs.SnapDesktopFilesDir, but explicitly denies access to all other snaps'
 // desktop files since xdg libraries may try to read all the desktop files
 // in the dir, causing excessive noise. (LP: #1868051)
-func getDesktopFileRules(snapInstanceName string) []string {
+func getDesktopFileRules(snapInstanceName string, spec *apparmor.Specification) []string {
+	// desktop-launch allows to read all .desktop files; but "deny" rules overrule any "allow"
+	// rule, so we must not add these rules if this snap uses the desktop-launch interface.
+	// Also, for security reasons, all these rules are removed if the desktop-launch interface
+	// is listed, thus only if it is really connected will the snap have any kind of access to
+	// these folders/files.
+	if spec != nil {
+		for _, plug := range spec.SnapAppSet().Info().Plugs {
+			if plug.Interface == "desktop-launch" {
+				return nil
+			}
+		}
+	}
```

Comment:
No, it shouldn't. It just checks if the snap uses the "desktop-launch" interface, but not if it is connected or not. As I comment in the description, it's not a security problem because, without these rules, no access is granted to the `XXXXX/applications` folders (while with these rules, you can read the list of files, but can't read the contents of the files), so a snap that includes the "desktop-launch" interface but doesn't connect it will have less rights than a snap that doesn't include it.